Source/UI/NewUI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -1,32 +1,83 @@
 import UIKit
 import Checkout
 
+private struct CardDetails {
+  var number: String?
+  var expiryDate: ExpiryDate?
+  var name: String?
+  var cvv: String?
+  var billingAddress: Address?
+  var phone: Phone?
+
+  init(number: String? = nil, expiryDate: ExpiryDate? = nil, name: String? = nil, cvv: String? = nil, billingAddress: Address? = nil, phone: Phone? = nil) {
+    self.number = number
+    self.expiryDate = expiryDate
+    self.name = name
+    self.cvv = cvv
+    self.billingAddress = billingAddress
+    self.phone = phone
+  }
+
+  func getCard() -> Card? {
+    guard let number = number, let expiryDate = expiryDate else { return nil }
+    return Card(number: number,
+                expiryDate: expiryDate,
+                name: name,
+                cvv: cvv,
+                billingAddress: billingAddress,
+                phone: phone)
+  }
+}
```

Comment:
Not sure if we want to have objects private on a file? I have previously mentioned creating a helper internal object that keeps your state, like `CardBuilder` or similar.

Reply:
please suggest code

Reply:
Do you know a way of suggesting GitHub changes where we move objects in new files? Otherwise all I could suggest is:

```suggestion

```

But I'm not sure its very helpful

Reply:
FYI, I've [found original suggestion](https://github.com/checkout/frames-ios/pull/241#discussion_r939879279)